### DISCUSSION TITLE
Which one to use for repeated pattern separated with a fixed part like 000-001-002 ?

### DISCUSSION THREAD

**Discussion by twnaing** [uNmAnNeR/imaskjs#909] [Q&A] [answered]
I want to mask the input that accepts three-digits input multiple times separated by dash as a fixed part (like in pattern).

`000-001-002-003-004`

Currently I am using RegExp mask with the following syntax `/^\d{0,3}(-\d{0,3})*$/`. But the approach 

- allows typing of 2 digits only
- does not auto add fixed part

**Comment by uNmAnNeR** [comment]
regexp is not appropriate for your case. You need to use pattern mask.

**Reply by twnaing** [reply]
As far as I understand from the documentation, pattern mask is fixed in size, right?

> Use pattern when:
> - ...
> - mask is fixed in size (optional symbols can provide some flexibility)

What I came up is `000[-000]`. As far as I understand it will allow up to 2 3-digits parts, not multiple.

So, my option is to repeat the optional pattern as long as the user might enter. Is my assumption correct?

**Reply by uNmAnNeR** [reply, marked as answer]
yes, that's the easiest way. But if you need endless mask you can also use dynamic mask with dispatch callback where you can generate new mask each time according to the existing value / applied input.

**Reply by twnaing** [reply]
Thank you for your guidance.

**Comment by twnaing** [comment]
I came up with this to achieve the title.

```javascript
  const IMOptions = {
    mask: [{ mask: "000" }],
    dispatch: function (appended, masked) {
      const number = (masked.value + appended).replace(/\D/g, "");
      const len = number.length;

      if (len < 3) {
        return masked.compiledMasks[0];
      }

      let times = parseInt(len / 3);
      if (len % 3 > 0) {
        times += 1;
      }

      return new MaskedPattern({
        mask: Array.from({ length: times }, () => "000").join("{-}"),
      });
    },
  };
```

**Reply by uNmAnNeR** [reply]
@twnaing looks good. if you don't mind i will add your case to the examples.

**Reply by twnaing** [reply]
Thank you. I am glad the code will be included in the examples.
